src/overlays/actors/ovl_Bg_Ladder/z_bg_ladder.c

```diff
@@ -19,22 +37,130 @@ const ActorInit Bg_Ladder_InitVars = {
     (ActorFunc)BgLadder_Init,
     (ActorFunc)BgLadder_Destroy,
     (ActorFunc)BgLadder_Update,
-    (ActorFunc)BgLadder_Draw
+    (ActorFunc)BgLadder_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Ladder_0x80A4D9F0/BgLadder_Init.asm")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+static Gfx* sLadderDLists[] = {
+    object_ladder_12rung_dlist,
+    object_ladder_16rung_dlist,
+    object_ladder_20rung_dlist,
+    object_ladder_24rung_dlist,
+};
+
+void BgLadder_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgLadder* this = THIS;
+    BgLadderSize size;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+
+    // `params` is treated as two bytes: switchFlag, size
+    // Has to be `thisx->params` to match
+    this->switchFlag = (thisx->params >> 8) & 0xFF;
+    thisx->params &= 0xFF;
```

Comment:
Would this match as `thisx->params = thisx->params & 0xFF`? Trying to see if we can convert this one to a macro as well.